plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -400,15 +403,12 @@ public List<MongoIndex> getIndexes(String schemaName, String tableName)
 
     public MongoCursor<Document> execute(MongoTableHandle tableHandle, List<MongoColumnHandle> columns)
     {
-        Document output = new Document();
-        for (MongoColumnHandle column : columns) {
-            output.append(column.getName(), 1);
-        }
         MongoCollection<Document> collection = getCollection(tableHandle.getSchemaTableName());
-        Document query = buildQuery(tableHandle.getConstraint());
-        FindIterable<Document> iterable = collection.find(query).projection(output);
+        Document projection = buildProjection(tableHandle, columns);
+        Document filter = buildFilter(tableHandle);
```

Comment:
Can we try to perform an `AND` of the filter we inject via PTF and the filter we apply in SQL. Something like this 

```
        Document query = tableHandle.getQuery().map(Document::parse).orElseGet(Document::new);
        query.putAll(buildQuery(tableHandle.getConstraint()));
```

This would apply both the filter when scanning the collection  and limit push down can also be implemented ?